surface_potential_analysis/kernel/kernel.py

```diff
@@ -470,3 +475,27 @@ def get_isotropic_kernel_from_operators(
         as_diagonal_noise_operators_from_full(operators),
         assert_isotropic=assert_isotropic,
     )
+
+
+def outer_product(*arrays):
+    grids = np.meshgrid(*arrays, indexing="ij")
+    return np.prod(grids, axis=0)
+
+
+def get_full_kernel_from_independent(
```

Comment:
move to build.py. Be more consistent in naming these kernels - are they 'inependent kernels' or 'seperate kernels'. rename get_separate_gaussian_isotropic_noise_kernel accordingly. I would maybe call them something like axis kernels